lib/puppet/provider/f5_datagroup/rest.rb

```diff
@@ -0,0 +1,101 @@
+require 'puppet/provider/f5'
+require 'json'
+
+Puppet::Type.type(:f5_datagroup).provide(:rest, parent: Puppet::Provider::F5) do
+
+  def self.instances
+    instances = []
+    dgroups = Puppet::Provider::F5.call_items('/mgmt/tm/ltm/data-group/internal')
+    return [] if dgroups.nil?
+
+    dgroups.each do |dgroup|
+      full_path_uri = dgroup['fullPath'].gsub('/','~')
+
+      instances << new(
+        ensure:                   :present,
+        name:                     dgroup['fullPath'],
+        description:              dgroup['description'],
+        type:                     dgroup['type'],
+        records:                  dgroup['records'],
+      )
+    end
+
+    instances
+  end
+
+  def self.prefetch(resources)
+    dgroups = instances
+    resources.keys.each do |name|
+      if provider = dgroups.find { |dgroup| dgroup.name == name }
+        resources[name].provider = provider
+      end
+    end
+  end
+
+  def create_message(basename, hash)
+    # Create the message by stripping :present.
+    new_hash            = hash.reject { |k, _| [:ensure, :provider, Puppet::Type.metaparams].flatten.include?(k) }
+    new_hash[:name]     = basename
+
+    return new_hash
+  end
+
+
+  def message(object)
+    # Allows us to pass in resources and get all the attributes out
+    # in the form of a hash.
+    message = object.to_hash
+
+    # Map for conversion in the message.
+    map = {
+    }
+
+    message = strip_nil_values(message)
+    message = convert_underscores(message)
+    #message = gen_sflow(message)
```

Comment:
There are 13 instances of commented out `message = gen_sflow(message)` in this PR.  The only place it should be is in the f5_vlan provider?

Reply:
Updated for all 13 instances/providers